src/dsl.jl

```diff
@@ -283,6 +286,17 @@ function extract_metadata(metadata_line::Expr)
     return metadata
 end
 
+
+
+struct UndeclaredSymbolicError <: Exception
+    msg::String
+end
+
+function Base.showerror(io::IO, err::UndeclaredSymbolicError) 
+    print(io, "UndeclaredSymbolicError: ")
+    print(io, err.msg)
+end
```

Comment:
Is there a reason we are creating a new error types for these? Don't really hurt, but also seems a bit... Overkill

Reply:
Yes, we can be a bit more explicit in the tests with catching it.

Reply:
Plus the name is more informative to a user about what the error is.

Reply:
Sounds good. (long term) we probably should be better at errors, both creating designated ones, but also using `throw` instead of just `error`.